File: 0064. Minimum Path Sum.cpp
class Solution
{
public:
    int minPathSum(vector<vector<int>>& grid)
    {
        for (int i = 0; i < grid.size(); ++i)
        {
            for (int j = 0; j < grid[i].size(); ++j)
            {
                if (i != 0 and j != 0)
                {
                    grid[i][j] += min(grid[i - 1][j], grid[i][j - 1]);
                }
                else if (j != 0)
                {
                    grid[i][j] += grid[i][j - 1];
                }
                else if (i != 0)
                {
                    grid[i][j] += grid[i - 1][j];
                }
            }
        }

        return grid[grid.size() - 1][grid[0].size() - 1];
    }
};

File: 0064. Minimum Path Sum.cs
public class Solution
{
    public int MinPathSum(int[][] grid)
    {
        for (int i = 0; i < grid.Length; ++i)
        {
            for (int j = 0; j < grid[i].Length; ++j)
            {
                var minPath = int.MaxValue;

                if (i != 0 && j != 0)
                {
                    grid[i][j] += Math.Min(grid[i - 1][j], grid[i][j - 1]);
                }
                else if (i != 0)
                {
                    grid[i][j] += grid[i - 1][j];
                }
                else if (j != 0)
                {
                    grid[i][j] += grid[i][j - 1];
                }
            }
        }

        return grid[^1][^1];
    }
}